src/main/java/de/tum/in/www1/artemis/service/TextAssessmentService.java

```diff
@@ -46,17 +42,11 @@ public TextAssessmentService(UserRepository userRepository, ComplaintResponseSer
      */
     public void prepareSubmissionForAssessment(TextSubmission textSubmission, @Nullable Result result) {
         final Participation participation = textSubmission.getParticipation();
-        final TextExercise exercise = (TextExercise) participation.getExercise();
-
-        final boolean computeFeedbackSuggestions = automaticTextFeedbackService.isPresent() && exercise.isAutomaticAssessmentEnabled();
```

Comment:
Feedback suggestions are now fetched asynchronously after a tutor has opened the submission. Before that, they were directly created before the submission could be opened. I changed this behavior because of performance concerns (waiting for an additional network request). In case it was particularly slow, I prevent the suggestions from appearing if the tutor adds feedback before the suggestions are incoming (see client code).

Reply:
Therefore, the `prepareSubmissionForAssessment` method does not need to know whether to compute feedback suggestions